kernel/data-pipeline/dialect/opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/OpenGaussWALDumper.java

```diff
@@ -80,15 +85,22 @@ public OpenGaussWALDumper(final DumperConfiguration dumperConfig, final IngestPo
         this.decodeWithTX = dumperConfig.isDecodeWithTX();
     }
     
-    @SneakyThrows(InterruptedException.class)
     @Override
     protected void runBlocking() {
+        while (reconnectTimes.get() <= 3) {
+            connect();
+        }
```

Comment:
Looks it will always run `connect()` several times.

Reconnect should just run on exception occur.